swift/Sources/FlatBuffers/ByteBuffer.swift

```diff
@@ -352,7 +352,11 @@ public struct ByteBuffer {
   ///   - position: the index of the object in the buffer
   @inline(__always)
   public func read<T>(def: T.Type, position: Int) -> T {
+#if swift(<5.7)
     _storage.memory.advanced(by: position).load(as: T.self)
+#else
+    _storage.memory.advanced(by: position).loadUnaligned(as: T.self)
```

Comment:
I'm not sure why this is needed in the read function? Shouldn't this ByteBuffer just be provided an aligned piece of memory at construction?

Reply:
It can be read from a network memory buffer or mmaped in on an unaligned address - to use the more robust load unaligned supports both the case when the buffer is aligned or not- thus avoiding unnecessary copies or forcing unpredictable padding on disk or the wire.

Reply:
This was inspired by the `swift-nio` and how they handled read and write into the memory. and since we dont want to expose the APIs for the `_storage.memory` for manipulation we hide it completely. Which is kinda what most struct classes do in swift, where they have the public API and then a hidden API to unsure people use it the way it's designed to.

Reply:
I don't think it is a good change. Explained here: https://github.com/google/flatbuffers/issues/7640#issuecomment-1318163581